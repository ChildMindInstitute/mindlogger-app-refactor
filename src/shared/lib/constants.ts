import { Platform } from 'react-native';

export const IS_IOS = Platform.OS === 'ios';
export const IS_ANDROID = Platform.OS === 'android';

export const AppVersion = '0.0.1';

export const ONE_MINUTE = 1000 * 60;
export const ONE_HOUR = ONE_MINUTE * 60;
