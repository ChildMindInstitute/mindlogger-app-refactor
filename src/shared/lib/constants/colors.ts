export const colors = {
  white: '#FFFFFF',
  whiteTsp: 'rgba(255, 255, 255, 0.8)',
  primary: '#0067A0',
  secondary: '#FFFFFF',
  secondary_50: '#ffffff',
  tertiary: '#404040',
  grey: '#808080',
  grey2: '#a0a0a0',
  darkGrey: '#3b3a3a',
  darkGrey2: '#6b6a6a',
  darkerGrey3: '#7E7E7E',
  darkerGrey: '#333333',
  mediumGrey: '#B6B6B6',
  lighterGrey: '#e8e8e8',
  lighterGrey2: '#f7f7f7',
  lighterGrey3: '#e2f0fe',
  lightGrey: '#00000026',
  lightGrey2: '#00000032',
  alert: '#e63232',
  blue: '#005fa3',
  blue2: '#24A3FF',
  lightBlue: '#dbf2ff',
  darkBlue: 'darkblue',
  green: '#008800',
  lightGreen: '#35FDB5',
  yellow: '#FFBD32',
  aqua: '#7FF7FF',
  black: '#000',
  red: '#FF0000',
  lightRed: '#ffdddd',
};
