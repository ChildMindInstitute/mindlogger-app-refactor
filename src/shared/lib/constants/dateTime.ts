export const MS_IN_MINUTE = 60000;
export const SECONDS_IN_MINUTE = 60;
export const MINUTES_IN_HOUR = 60;

const date = new Date();
date.setHours(0);
date.setMinutes(0);
date.setSeconds(0);
export const MIDNIGHT_DATE = date;
