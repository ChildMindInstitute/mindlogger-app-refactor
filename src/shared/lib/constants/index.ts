import { Platform } from 'react-native';

import Config from 'react-native-config';
import { CameraOptions, ImageLibraryOptions } from 'react-native-image-picker';
import { PERMISSIONS } from 'react-native-permissions';

import { Language } from '../types';

export * from './colors';
export * from './dateTime';

export const IS_IOS = Platform.OS === 'ios';
export const IS_ANDROID = Platform.OS === 'android';
export const IS_ANDROID_12_OR_HIGHER =
  IS_ANDROID && (Platform.Version as number) >= 31;

export const ENV = Config.ENV;
export const API_URL = Config.API_URL as string;
export const STORE_ENCRYPTION_KEY = Config.STORE_ENCRYPTION_KEY;

// @ts-ignore
export const APP_VERSION = process.env.VERSION;

export const ONE_SECOND = 1000;
export const ONE_MINUTE = ONE_SECOND * 60;
export const ONE_HOUR = ONE_MINUTE * 60;

export const DEFAULT_LANGUAGE: Language = 'en';

export const LOCATION_PERMISSIONS = Platform.select({
  android: PERMISSIONS.ANDROID.ACCESS_FINE_LOCATION,
  ios: PERMISSIONS.IOS.LOCATION_WHEN_IN_USE,
});

export const CAMERA_PERMISSIONS = Platform.select({
  android: PERMISSIONS.ANDROID.CAMERA,
  ios: PERMISSIONS.IOS.CAMERA,
});

export const GALLERY_ANDROID_PERMISSIONS = [
  PERMISSIONS.ANDROID.WRITE_EXTERNAL_STORAGE,
  PERMISSIONS.ANDROID.READ_EXTERNAL_STORAGE,
];

export const GALLERY_IOS_PERMISSIONS = PERMISSIONS.IOS.PHOTO_LIBRARY;

export const GALLERY_PHOTO_OPTIONS: ImageLibraryOptions = {
  mediaType: 'photo',
  quality: 0.8,
  maxWidth: 800,
  maxHeight: 800,
  selectionLimit: 1,
};

export const GALLERY_VIDEO_OPTIONS: ImageLibraryOptions = {
  mediaType: 'video',
  videoQuality: 'high',
  quality: 0.9,
  maxWidth: 800,
  maxHeight: 800,
};

export const VIDEO_RECORD_OPTIONS: CameraOptions = {
  mediaType: 'video',
  cameraType: 'back',
  videoQuality: 'high',
  durationLimit: 60,
  quality: 0.9,
  saveToPhotos: true,
};

export const PHOTO_TAKE_OPTIONS: CameraOptions = {
  mediaType: 'photo',
  maxWidth: 800,
  maxHeight: 800,
  quality: 0.9,
  saveToPhotos: true,
};

export const MICROPHONE_PERMISSIONS = Platform.select({
  android: PERMISSIONS.ANDROID.RECORD_AUDIO,
  ios: PERMISSIONS.IOS.MICROPHONE,
});

export const DAYS_OF_WEEK_NUMBERS = [0, 1, 2, 3, 4, 5, 6];
export const DAYS_OF_WEEK_SHORT_NAMES = ['M', 'T', 'W', 'T', 'F', 'S', 'S'];
