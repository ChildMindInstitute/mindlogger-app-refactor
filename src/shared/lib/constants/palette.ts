import { hexToRgba } from '../utils/theme.utils';

export interface Palette {
  [key: string]: string;
}

/**
 * This references all base "primitive" colors.
 * These colors will be used to give values to
 * semantic colors, like "on_container".
 */
const basePalette = {
  primary: '#0B0907',
  primary0: '#000',
  primary10: '#1D1B19',
  primary20: '#32302D',
  primary30: '#484744',
  primary40: '#5F5E5B',
  primary50: '#787773',
  primary60: '#91918E',
  primary70: '#ACABA9',
  primary80: '#C6C6C5',
  primary90: '#E3E2E1',
  primary95: '#F1F0EF',
  primary99: '#FDFCFC',
  primary100: '#FFF',
  secondary: '#004CED',
  secondary0: '#000',
  secondary10: '#041847',
  secondary20: '#182D65',
  secondary30: '#173F9F',
  secondary40: '#004CED',
  secondary50: '#216DFF',
  secondary60: '#3C90FF',
  secondary70: '#55B1FF',
  secondary80: '#8BCEFF',
  secondary90: '#CAE6FF',
  secondary95: '#E5F3FF',
  secondary99: '#FBFCFF',
  secondary100: '#FFF',
  tertiary: '#944262',
  tertiary0: '#000',
  tertiary10: '#320F1D',
  tertiary20: '#4B2432',
  tertiary30: '#6B364A',
  tertiary40: '#944262',
  tertiary50: '#AF5C82',
  tertiary60: '#CA779E',
  tertiary70: '#D69AB8',
  tertiary80: '#E5BCD1',
  tertiary90: '#F0DAE5',
  tertiary95: '#F9EEF4',
  tertiary99: '#FFFBFF',
  tertiary100: '#FFF',
  error: '#BA1A1A',
  error0: '#000',
  error10: '#410002',
  error20: '#690004',
  error30: '#930009',
  error40: '#BA1A1A',
  error50: '#DE3730',
  error60: '#FF5449',
  error70: '#FE897D',
  error80: '#FFB4AB',
  error90: '#FFDAD6',
  error95: '#FFEDEA',
  error99: '#FFFBFF',
  error100: '#FFF',
  neutral: '#5F5E5B',
  neutral0: '#000',
  neutral10: '#1D1B19',
  neutral20: '#32302D',
  neutral30: '#484744',
  neutral40: '#5F5E5B',
  neutral50: '#787773',
  neutral60: '#91918E',
  neutral70: '#ACABA9',
  neutral80: '#C6C6C5',
  neutral90: '#E3E2E1',
  neutral95: '#F1F0EF',
  neutral99: '#FDFCFC',
  neutral100: '#FFF',
  neutral_variant: '#5F5E5B',
  neutral_variant0: '#000',
  neutral_variant10: '#1D1B19',
  neutral_variant20: '#32302D',
  neutral_variant30: '#484744',
  neutral_variant40: '#5F5E5B',
  neutral_variant50: '#787773',
  neutral_variant60: '#91918E',
  neutral_variant70: '#ACABA9',
  neutral_variant80: '#C6C6C5',
  neutral_variant90: '#E3E2E1',
  neutral_variant95: '#F1F0EF',
  neutral_variant99: '#FDFCFC',
  neutral_variant100: '#FFF',
  surface: '#FDFCFC',
  surface1: '#F5F4F4',
  surface2: '#F0EFEF',
  surface3: '#ECEBEA',
  surface4: '#EAE9E9',
  surface5: '#E7E6E5',
  blue: '#0152FD',
  blue_light: '#B3CBFE',
  brown: '#79403C',
  brown_light: '#D6C6C4',
  gray: '#787773',
  gray_light: '#D6D6D5',
  green: '#386348',
  green_light: '#C3D0C8',
  orange: '#E65838',
  orange_light: '#F7CDC3',
  pink: '#A75276',
  pink_light: '#E6CCD7',
  yellow: '#DAB417',
  yellow_light: '#F4E8BA',
  purple: '#6C4E9B',
  purple_light: '#D3CAE1',
  red: '#B83236',
  red_light: '#EAC1C3',
  white: '#FFFFFF',
} as const;

/**
 * This contains all flattened palette colors with its tonal variants,
 * as well as all semantic colors (on_container, container, etc).
 */
export const semanticPalette = {
  success: basePalette.green,
  warning: basePalette.yellow,
  info: basePalette.blue,
  on_surface: basePalette.neutral10,
  surface_variant: basePalette.neutral90,
  on_surface_variant: basePalette.neutral_variant30,
  on_primary: basePalette.primary100,
  outline: basePalette.neutral_variant50,
  outline_variant: basePalette.neutral_variant80,
  secondary_container: basePalette.secondary90,

  // TODO: Use proper semantic names for remaining colors
  secondary: '#FFFFFF',
  secondary_50: '#ffffff',
  tertiary: '#404040',
  grey: '#808080',
  grey2: '#a0a0a0',
  grey3: '#bbb',
  grey4: '#72777F',
  darkon_surface: '#E2E2E5',
  darkGrey: '#3b3a3a',
  darkGrey2: '#6b6a6a',
  darkerGrey3: '#7E7E7E',
  darkerGrey4: '#42474E',
  darkerGreyBackground: '#7E7E7E80',
  darkerGrey: '#333333',
  darkerGrey2: '#101010',
  mediumGrey: '#B6B6B6',
  lighterGrey0: '#E0E0E0',
  lighterGrey2: '#f7f7f7',
  lighterGrey3: '#e2f0fe',
  lighterGrey5: '#FFE2E2',
  lighterGrey6: '#D5E4F7',
  lighterGrey7: '#DEE3EB',
  lighterGrey8: '#e0ebf4',
  lightGrey: '#00000026',
  lightGrey2: '#00000032',
  lightGrey3: '#D2E2EC4D',
  lightGrey4: '#F3F5F4',
  alert: '#e63232',
  alertLight: '#FFCCCC',
  alertDark: '#93000A',
  blue: '#005fa3',
  blue2: '#24A3FF',
  blue3: '#00639A',
  blue4: '$02649B',
  lightBlue: '#dbf2ff',
  lightBlue2: '#D8E4ED',
  darkBlue: 'darkblue',
  green: '#008800',
  lightGreen: '#35FDB5',
  lightGreen2: '#57ba79',
  yellow: '#FFBD32',
  lightYellow: '#F2E7B8',
  dimYellow: '#D6AF35',
  aqua: '#7FF7FF',
  black: '#000',
  red: '#FF0000',
  red2: '#D33134',
  lightRed: '#ffdddd',
  codePink: '#d63384',
  errorRed: '#800000',
} as const;

/**
 * These are alpha variants of the semantic colors.
 * They exist separately from the semantic palette for
 * better maintainability.
 */
const alphaVariantsPalette = {
  success_container: hexToRgba(semanticPalette.success, 0.3),
  error_container: hexToRgba(basePalette.error, 0.3),
  warning_container: hexToRgba(semanticPalette.warning, 0.3),
  info_container: hexToRgba(semanticPalette.info, 0.3),
  white_alpha80: hexToRgba(basePalette.white, 0.8),
  white_alpha60: hexToRgba(basePalette.white, 0.6),
  on_surface_alpha12: hexToRgba(basePalette.neutral10, 0.12),
  on_surface_alpha30: hexToRgba(basePalette.neutral10, 0.3),
  toast_container: hexToRgba(basePalette.primary, 0.8),
} as const;

export const palette = {
  ...basePalette,
  ...semanticPalette,
  ...alphaVariantsPalette,
} as const satisfies Palette;
