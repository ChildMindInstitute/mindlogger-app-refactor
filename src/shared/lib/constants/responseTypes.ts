import { ResponseType } from '@app/shared/api/services/ActivityItemDto';

/**
 * NOTE: When updating this file, make sure to make corresponding changes in
 * the Web & Admin apps:
 * - https://github.com/ChildMindInstitute/mindlogger-admin/blob/main/src/shared/utils/responseType.ts
 */

export const universalSupportedResponseTypes: ResponseType[] = [
  'audioPlayer',
  'date',
  'message',
  'multiSelect',
  'multiSelectRows',
  'numberSelect',
  'paragraphText',
  'singleSelect',
  'singleSelectRows',
  'slider',
  'sliderRows',
  'text',
  'time',
  'timeRange',
];

export const appSupportedResponseTypes: ResponseType[] = [
  'ABTrails',
  'audio',
  'drawing',
  'flanker',
  'geolocation',
  'photo',
  'stabilityTracker',
  'video',
];

export const webSupportedResponseTypes: ResponseType[] = ['phrasalTemplate'];
