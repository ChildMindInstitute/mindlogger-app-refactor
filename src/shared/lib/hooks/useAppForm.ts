import {
  useForm,
  UseFormProps,
  SubmitHandler,
  SubmitErrorHandler,
} from 'react-hook-form';
import { z } from 'zod';

import { zodResolver } from '../validation/zodResolver';

export const useAppForm = <Schema extends z.ZodTypeAny>(
  schema: Schema,
  props: UseFormProps<z.input<Schema>> & {
    onSubmitSuccess: SubmitHandler<z.output<Schema>>;
    onSubmitFail?: SubmitErrorHandler<z.input<Schema>>;
  },
) => {
  const form = useForm({
    ...props,
    resolver: zodResolver(schema),
  });

  const submit = form.handleSubmit(props.onSubmitSuccess, props.onSubmitFail);

  return { form, submit };
};
