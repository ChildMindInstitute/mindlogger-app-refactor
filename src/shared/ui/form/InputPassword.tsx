import { FC, useCallback, useState } from 'react';
import { TouchableWithoutFeedback } from 'react-native';

import {
  Controller,
  useController,
  useFormContext,
  useWatch,
} from 'react-hook-form';

import { validatePassword } from '@app/features/sign-up/validation/SignUpFormSchema';
import PasswordRequirements from '@shared/ui/form/PasswordRequirements';
import Input from '@shared/ui/Input';
import { InputIcon } from '@shared/ui/InputIcon';

import { InputFieldProps } from './InputField';
import { EyeIcon, EyeSlashIcon } from '../icons';

export const InputPassword: FC<InputFieldProps> = ({
  name,
  defaultValue = '',
  placeholder,
  mode = 'light',
  backgroundColor,
  ...props
}) => {
  const { control } = useFormContext();
  const [displayHints, setDisplayHints] = useState(false);
  const [isPasswordHidden, setPasswordHidden] = useState(true);
  const {
    field: { onChange: onFormChange, value, ref, onBlur },
    fieldState: { error },
  } = useController({
    name,
    control,
    defaultValue,
  });

  const { password } = useWatch({
    control,
  });

  const checkPassword = useCallback(() => {
    const { minCharacters, noSpaces, specialCharacters, upperCase } =
      validatePassword(password);
    const toDisplay = [minCharacters, noSpaces, specialCharacters, upperCase];
    return {
      isEmpty: !!error,
      requirements: toDisplay,
      isInvalid: password.length > 0 && toDisplay.some(r => !r.isValid),
    };
  }, [password, error]);

  const { requirements, isInvalid } = checkPassword();
  const ShowPasswordIcon = isPasswordHidden ? EyeSlashIcon : EyeIcon;

  return (
    <>
      <Controller
        control={control}
        render={() => (
          <InputIcon
            input={
              <Input
                ref={ref}
                onFocus={() => setDisplayHints(true)}
                onBlur={() => {
                  setDisplayHints(password.length > 1);
                  onBlur();
                }}
                onChangeText={onFormChange}
                value={value}
                placeholder={placeholder}
                autoCapitalize="none"
                mode={mode}
                backgroundColor={backgroundColor}
                secureTextEntry={isPasswordHidden}
                {...props}
                isInvalid={isInvalid || !!error}
              />
            }
            icon={
              <TouchableWithoutFeedback
                onPress={() => setPasswordHidden(!isPasswordHidden)}
              >
                <ShowPasswordIcon size={24} color={mode} />
              </TouchableWithoutFeedback>
            }
          />
        )}
        name={name}
      />

      {(displayHints || !!error) && (
        <PasswordRequirements requirements={requirements} />
      )}
    </>
  );
};

export default InputPassword;
