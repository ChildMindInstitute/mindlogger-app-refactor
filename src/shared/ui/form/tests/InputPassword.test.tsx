import React from 'react';
import { TextInput } from 'react-native';

import { FormProvider } from 'react-hook-form';
import renderer, { act } from 'react-test-renderer';

import TamaguiProvider from '@app/app/ui/AppProvider/TamaguiProvider';
import SignUpFormSchema from '@app/features/sign-up/validation/SignUpFormSchema';
import { colors } from '@app/shared/lib/constants/colors';
import useAppForm from '@app/shared/lib/hooks/useAppForm';

import InputPassword from '../InputPassword';

/* Test helper functions */

const getComponentByTestId = (
  instance: renderer.ReactTestInstance,
  testID: string,
) => {
  return instance.find(c => c.props.testID === testID).props;
};

const getPasswordRequirement = (
  instance: renderer.ReactTestInstance,
  requirementID: string,
) => {
  const requirement = getComponentByTestId(
    instance,
    `password_requirements:${requirementID}`,
  );
  return {
    icon: requirement.children[0].props.children,
    text: requirement.children[1].props.children,
  };
};

const getPasswordRequirements = (instance: renderer.ReactTestInstance) => {
  const minCharacters = getPasswordRequirement(instance, 'at_least_characters');

  const capitalLetter = getPasswordRequirement(
    instance,
    'at_least_capital_letter',
  );

  const specialCharacters = getPasswordRequirement(
    instance,
    'no_special_characters',
  );
  const blankSpaces = getPasswordRequirement(instance, 'no_blank_spaces');

  return { minCharacters, capitalLetter, specialCharacters, blankSpaces };
};

const FormTestWrapper = ({
  children,
  defaultValues = {},
}: {
  children: React.ReactNode;
  defaultValues: any;
}) => {
  const { form } = useAppForm(SignUpFormSchema, {
    defaultValues: {
      email: '',
      firstName: '',
      lastName: '',
      password: '',
      ...defaultValues,
    },
    onSubmitSuccess: () => {},
  });
  return (
    <TamaguiProvider>
      <FormProvider {...form}>{children}</FormProvider>
    </TamaguiProvider>
  );
};

const createTest = ({
  defaultValues = {},
  displayRequirements = true,
}: any = {}) => {
  const tree = renderer.create(
    <FormTestWrapper defaultValues={defaultValues}>
      <InputPassword name="password" placeholder="Password" />
    </FormTestWrapper>,
  );
  const instance = tree.root;
  if (displayRequirements) {
    const mEvent = { target: { value: '' } };
    act(() => {
      const input: any = instance.findByType(TextInput);
      input.props.onFocus(mEvent);
    });
  }
  return { tree, instance };
};

/* Test use-cases */

describe('InputPassword', () => {
  it('should render properly without displaying the password requirements', () => {
    const { tree } = createTest({ displayRequirements: false });
    expect(tree).toMatchSnapshot();
  });

  it('should render properly displaying the password requirements', () => {
    const { tree } = createTest({ displayRequirements: true });
    expect(tree).toMatchSnapshot();
  });

  it('should display validation when the password have 8 characters', () => {
    expect(
      getPasswordRequirements(
        createTest({
          defaultValues: {
            password: 'testpassword',
          },
        }).instance,
      ).minCharacters.icon.props.color,
    ).toBe(colors.white);
    expect(
      getPasswordRequirements(
        createTest({
          defaultValues: {
            password: '',
          },
        }).instance,
      ).minCharacters.icon.props.color,
    ).toBe(colors.whiteTsp2);
  });

  it('should display validation when the password have no blank spaces', () => {
    expect(
      getPasswordRequirements(
        createTest({
          defaultValues: {
            password: 'a',
          },
        }).instance,
      ).blankSpaces.icon.props.color,
    ).toBe(colors.white);
    expect(
      getPasswordRequirements(
        createTest({
          defaultValues: {
            password: 'a ',
          },
        }).instance,
      ).blankSpaces.icon.props.color,
    ).toBe(colors.whiteTsp2);
  });

  it('should display validation when the password have special characters', () => {
    expect(
      getPasswordRequirements(
        createTest({
          defaultValues: {
            password: 'testpassword@',
          },
        }).instance,
      ).specialCharacters.icon.props.color,
    ).toBe(colors.white);
    expect(
      getPasswordRequirements(
        createTest({
          defaultValues: {
            password: 'testpassword',
          },
        }).instance,
      ).specialCharacters.icon.props.color,
    ).toBe(colors.whiteTsp2);
  });

  it('should display validation when the password have a capital letter', () => {
    expect(
      getPasswordRequirements(
        createTest({
          defaultValues: {
            password: 'testPassword',
          },
        }).instance,
      ).capitalLetter.icon.props.color,
    ).toBe(colors.white);
    expect(
      getPasswordRequirements(
        createTest({
          defaultValues: {
            password: 'testpassword',
          },
        }).instance,
      ).capitalLetter.icon.props.color,
    ).toBe(colors.whiteTsp2);
  });
});
