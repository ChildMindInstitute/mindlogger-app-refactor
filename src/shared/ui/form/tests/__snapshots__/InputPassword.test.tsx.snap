// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputPassword should render properly displaying the password requirements 1`] = `
[
  <View
    style={
      {
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "height": "auto",
        }
      }
    >
      <TextInput
        autoCapitalize="none"
        color="rgba(255,255,255,1.00)"
        focusable={true}
        fontSize={20}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        outlineWidth={0}
        placeholder="Password"
        placeholderTextColor="rgba(255,255,255,0.60)"
        secureTextEntry={true}
        selectionColor="black"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        {
          "position": "absolute",
          "right": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectable={false}
        style={
          [
            {
              "color": "light",
              "fontSize": 24,
            },
            undefined,
            {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
  <span
    fontFamily="System"
    suppressHighlighting={true}
  >
    password_requirements:must_include
  </span>,
  <View>
    <View
      testID="password_requirements:at_least_characters"
    >
      <View>
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "rgba(255, 255, 255, 0.6)",
                "fontSize": 16,
              },
              undefined,
              {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
      <span
        fontFamily="System"
        suppressHighlighting={true}
      >
        password_requirements:at_least_characters
      </span>
    </View>
    <View
      testID="password_requirements:no_blank_spaces"
    >
      <View>
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "rgba(255, 255, 255, 0.6)",
                "fontSize": 16,
              },
              undefined,
              {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
      <span
        fontFamily="System"
        suppressHighlighting={true}
      >
        password_requirements:no_blank_spaces
      </span>
    </View>
    <View
      testID="password_requirements:no_special_characters"
    >
      <View>
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "rgba(255, 255, 255, 0.6)",
                "fontSize": 16,
              },
              undefined,
              {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
      <span
        fontFamily="System"
        suppressHighlighting={true}
      >
        password_requirements:no_special_characters
      </span>
    </View>
    <View
      testID="password_requirements:at_least_capital_letter"
    >
      <View>
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "rgba(255, 255, 255, 0.6)",
                "fontSize": 16,
              },
              undefined,
              {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
      <span
        fontFamily="System"
        suppressHighlighting={true}
      >
        password_requirements:at_least_capital_letter
      </span>
    </View>
  </View>,
]
`;

exports[`InputPassword should render properly without displaying the password requirements 1`] = `
<View
  style={
    {
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "height": "auto",
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      color="rgba(255,255,255,1.00)"
      focusable={true}
      fontSize={20}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      outlineWidth={0}
      placeholder="Password"
      placeholderTextColor="rgba(255,255,255,0.60)"
      secureTextEntry={true}
      selectionColor="black"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      {
        "position": "absolute",
        "right": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectable={false}
      style={
        [
          {
            "color": "light",
            "fontSize": 24,
          },
          undefined,
          {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
